### PR TITLE
OSLogin API.

### PR DESCRIPTION
**Pull request by lukesneeringer**
Assigning to @jbolinger since @jonparrott is out this week.
This should be launch ready.

### CODE REVIEW

**Comment by lukesneeringer** [comment]
Marking `do not merge` since there is a proto missing.

File: oslogin/docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+# The encoding of source files.
+#source_encoding = 'utf-8-sig'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'google-cloud-os-login'
+copyright = u'2017, Google'
```
Comment (chemelnucfin) [inline oslogin/docs/conf.py]: 2018?

File: oslogin/nox.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    session.run('py.test', '--quiet', os.path.join('tests', 'unit'))
+
+
+@nox.session
+def lint_setup_py(session):
+    """Verify that setup.py is valid (including RST check)."""
+    session.interpreter = 'python3.6'
+    session.install('docutils', 'pygments')
+    session.run('python', 'setup.py', 'check', '--restructuredtext',
+                '--strict')
```
Comment (chemelnucfin) [inline oslogin/nox.py]: Are cover and system still missing?

Comment (lukesneeringer) [reply oslogin/nox.py]: There are no autogen smoke tests for os-login.

File: oslogin/docs/conf.py (hunk 1 above)
Comment (lukesneeringer) [reply oslogin/docs/conf.py]: Thanks, I will bump that once I figure out how to generate the proto that did not come through.

**Comment by tseaver** [comment]
@lukesneeringer FWIW, I can't see the docs for the API yet on cloud.google.com.

**Comment by lukesneeringer** [comment]
Oh, hey, the docs/index.rst file needs its links updated since this one is weird. Nice catch, thanks.

**Comment by lukesneeringer** [comment]
Putting in a new PR for that.

**Pull request merged**